docs/architecture.rst

```diff
@@ -0,0 +1,191 @@
+2. **Static Assets**: Browser loads JavaScript, CSS, and images from static files
+3. **Data Loading**: JavaScript calls the appropriate Core's REST endpoint for data (e.g., to populate a table)
+4. **Interactive Updates**: Further interactions (filtering, row actions) go through REST → JS updates the DOM
+
+Types of Core
+-------------
+
+Django IS Core provides several types of Core classes that can be combined based on your needs:
+
+Which Core Type Should I Use?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
++---------------------------+---------------------------+---------------------------+
+| **Use Case**              | **Core Type**             | **What You Get**          |
++===========================+===========================+===========================+
+| Full admin interface      | DjangoUiRestCore          | UI views + REST API       |
++---------------------------+---------------------------+---------------------------+
+| REST API only             | DjangoRestCore            | API endpoints only        |
++---------------------------+---------------------------+---------------------------+
+| UI with external API      | DjangoUiCore              | UI views (REST elsewhere) |
++---------------------------+---------------------------+---------------------------+
+| Custom frontend (React)   | DjangoRestCore            | API for your frontend     |
++---------------------------+---------------------------+---------------------------+
+| Static pages only         | UiCore                    | Simple server-side views  |
++---------------------------+---------------------------+---------------------------+
+
+UI Core
+^^^^^^^
+
+Handles server-side rendered views without REST queries during interaction. Useful for simple, static pages.
+
+REST Core
+^^^^^^^^^
+
+Generates REST API over models providing CRUD operations, pagination, filtering, etc. Can be used independently to build custom frontends (e.g., React, Vue).
+
+Django UI Core
+^^^^^^^^^^^^^^
+
+A UI Core specialized for Django models providing default list/detail/create views automatically.
+
+Django REST Core
+^^^^^^^^^^^^^^^^
+
+Default REST API over Django models with automatic CRUD endpoints.
+
+Combined Core (DjangoUiRestCore)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The most common case - combines UI + REST in a single Core. Provides both the admin interface and the REST API that powers it.
+
+.. tip::
+   **Recommended**: Use ``DjangoUiRestCore`` for most use cases. It gives you both admin UI and REST API with a single class definition.
+
+Example::
+
+    from is_core.main import DjangoUiRestCore
+
+    class ArticleCore(DjangoUiRestCore):
+        model = Article
+        list_fields = ('title', 'author', 'created_at')
+        form_fields = ('title', 'content', 'author')
+
+Registration and Auto-discovery
+--------------------------------
+
+Django IS Core uses automatic registration through metaclasses and singletons:
```

Comment:
Specify, that the only thing that library users need to do is create `cores.py` in their application registred in `INSTALLED_APPS`.